external-crates/move/move-binary-format/src/deserializer.rs

```diff
@@ -572,7 +590,7 @@ fn build_common_tables(
                 load_constant_pool(binary, table, common.get_constant_pool())?;
             }
             TableType::METADATA => {
-                if binary.version() < VERSION_5 {
+                if binary.check_no_extraneous_bytes() || binary.version() < VERSION_5 {
```

Comment:
so this section is banned from now on, right?

Reply:
correct, it will give the error MALFORMED if it sees a TableType::METADATA